src/convertToObject.js

```diff
@@ -3,16 +3,30 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = {};
+
+  const propertyLines = sourceString.trim().split(';');
+
+  for (const line of propertyLines) {
+    const [key, value] = line.split(':').map(prop => prop.trim());
+
+    if (key && value) {
+      styles[key] = value;
+    }
+  }
```

Comment:
Let's try another way to solve this.
You can use `reduce()` with empty object as `initialValue` to create result object.

And don't forget about how to write chained methods. (Checklist p4)

```
const cssStyles = styles
 .map(style => style)
 .filter(style => style.length)
 .slice(0, 5)
```